File: code/timer2/python/timer_once.py
import threading
import datetime

def say_hello():
    print("{}: Hello, World!".format(datetime.datetime.now()))

# 创建一个 Timer，5 秒后执行 say_hello 函数
timer = threading.Timer(5.0, say_hello)

# 启动定时器
timer.start()

print("{}: Timer started, waiting for it to trigger...".format(datetime.datetime.now()))

File: code/threading.Timer/recipe-02/examples/timer_once.cpp
#include "timer.hpp"
#include "print_message.hpp"
#include <chrono>
#include <thread>

void say_hello() {
    print_message("Hello, World!");
}

int main() {
    auto timer = Timer{say_hello, std::chrono::seconds{3}};
    timer.start();
    print_message("Timer started, waiting for it to trigger...");
    while (timer.isActive()) {
        print_message("Timer is active...");
        std::this_thread::sleep_for(std::chrono::seconds{1});
    }
    print_message("Timer is not active.");
}
